spec/browsers/deezer_user_agent_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'user_agent'
+
+shared_examples 'Deezer' do |version, platform, os, type|
+  it "returns 'Deezer' as its browser" do
+    expect(useragent.browser).to eq('Deezer')
+  end
+
+  it "returns '#{version}' as its version" do
+    expect(useragent.version).to eq(version)
+  end
+
+  it "returns '#{platform}' as its platform" do
+    expect(useragent.platform).to eq(platform)
+  end
+
+  it "returns '#{os}' as its operating system" do
+    expect(useragent.os).to eq(os)
+  end
+
+  if type == :mobile
+    it { expect(useragent).to be_mobile }
+    it { expect(useragent).not_to be_desktop }
+  else
+    it { expect(useragent).to be_desktop }
+    it { expect(useragent).not_to be_mobile }
+  end
+
+  it { expect(useragent).not_to be_speaker }
+  it { expect(useragent).not_to be_bot }
+end
+
+describe "UserAgent: Deezer/6.2.28.73 (Android; 11; Mobile; fr) samsung SM-G973F" do
+  let(:useragent) { UserAgent.parse("Deezer/6.2.28.73 (Android; 11; Mobile; fr) samsung SM-G973F") }
+
+  it_behaves_like 'Deezer', '6.2.28.73', 'Android', 'Android 11', :mobile
+end
+
+describe "UserAgent: Deezer/6.0.10.201 (Android; 5.1.1; Tablet; nl) samsung SM-T280" do
+  let(:useragent) { UserAgent.parse("Deezer/6.0.10.201 (Android; 5.1.1; Tablet; nl) samsung SM-T280") }
+
+  it_behaves_like 'Deezer', '6.0.10.201', 'Android', 'Android 5.1.1', :mobile
+end
+
+describe "UserAgent: Deezer/8.39.0.6 CFNetwork/1220.1 Darwin/20.3.0" do
+  let(:useragent) { UserAgent.parse("Deezer/8.39.0.6 CFNetwork/1220.1 Darwin/20.3.0") }
+
+  it_behaves_like 'Deezer', '8.39.0.6', 'iOS', 'iOS 14.4.x', :mobile
+end
+
+describe "UserAgent: Deezer/4.34.10 (Electron; windows/10.0.19041; Desktop; de)" do
+  let(:useragent) { UserAgent.parse("Deezer/4.34.10 (Electron; windows/10.0.19041; Desktop; de)") }
+
+  it_behaves_like 'Deezer', '4.34.10', 'Windows', 'Windows 10.0.19041', :desktop
```

Comment:
A thought for a future release: we should probably translate the Windows 10 version numbers to marketing releases - e.g., `Windows 10.0.19041` is actually `Windows 10 Version 2004`